src/App.tsx

```diff
@@ -1,14 +1,32 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
-export class App extends React.Component<{}, State> {
-  state: State = {};
+export class App extends React.PureComponent<{}, State> {
+  state: State = {
+    query: '',
+  };
+
+  handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    const key = name as keyof State;
+
+    this.setState({ [key]: value });
+  };
 
   render() {
+    const { query } = this.state;
+    const visibleMovies: Movie[] = [...moviesFromServer].filter(movie => (
```

Comment:
create getVisibleMovies method
```suggestion
    const visibleMovies = this.getVisibleMovies(moviesFromServer);
```